nixos/modules/services/monitoring/osquery.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+with lib;
+
+let
+  cfg = config.services.osquery;
+  runtimeValue = flag: default:
+    cfg.config.options.${flag}
+      or cfg.flags.${flag}
+      or default;
+in
+{
+  options.services.osquery = {
+    enable = mkEnableOption (mdDoc "osqueryd daemon");
+
+    config = mkOption {
+      default = { };
+      description = mdDoc ''
+        Configuration to be written to the osqueryd JSON configuration file.
+        To understand the configuration format, refer to https://osquery.readthedocs.io/en/stable/deployment/configuration/#configuration-components.
+      '';
+      example = {
+        options.utc = false;
+      };
+      type = types.attrs;
+    };
+
+    flagfile = mkOption {
+      default = "/etc/osquery/osquery.flags";
+      description = mdDoc ''
+        Path to the flagfile used to provide CLI flags to osqueryd.
+        For more information, refer to https://osquery.readthedocs.io/en/stable/installation/cli-flags/#flagfile.
+      '';
+      type = types.path;
+    };
+    flags = mkOption {
+      default = { };
+      description = mdDoc ''
+        Attribute set of flag names and values to be written to the osqueryd flagfile.
+        For more information, refer to https://osquery.readthedocs.io/en/stable/installation/cli-flags.
+      '';
+      example = {
+        config_refresh = "10";
+      };
+      type = types.attrsOf types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
+      "osquery/osquery.conf".text = toJSON cfg.config;
+      "osquery/osquery.flags".text = concatStringsSep "\n" (mapAttrsToList (name: value: "--${name}=${value}") cfg.flags);
```

Comment:
Not to self, that the `flagfile` option should be the value here. Or perhaps we just remove the `flagfile` option when we put both of these in the Nix store.

Reply:
> Or perhaps we just remove the flagfile option when we put both of these in the Nix store.

Yeah, that's what I was getting at when I was interrogating whether we should have `flagfile` at all.

Reply:
I do use it in the tests so that `osqueryi` uses the same configuration. I'll need to find a solution for that too.

Reply:
The flag file is now in the Nix store.